### DISCUSSION TITLE
Laggy cursor in map select

### DISCUSSION THREAD

**Discussion by taep96** [ppy/osu#16458] [Q&A] [answered]
On a fresh install of Arch Linux I am experiencing low fps and cursor movement delay. That wasn't happening on my previous OS which was also Linux. On both I have used same latest version (2022.108.0). I have also tried using in-game, OTD and my own tablet driver which didn't affect anyting

**Comment by bdach** [comment]
no logs, no video, i can't reproduce. please provide actual details if you wish to receive help

**Comment by taep96** [comment, marked as answer]
I don't think a video would help. Screen recording will drop performance even further and there is nothing to compare to?
[logs.zip](https://github.com/ppy/osu/files/7875083/logs.zip)

**Reply by taep96** [reply]
@bdach Meant to reply, not create a new thread

**Reply by bdach** [reply]
the archive is empty. there are no logs there. just an empty logs folder.

to restate: i CANNOT help you if you don't provide me anything to go on. the description of the problem is vague enough wherein this could be anything. i need something concrete to go on. you could even use your phone to record the screen or something, that would be at least helpful.

**Reply by taep96** [reply]
Audio seems fine but fps feels smaller than before and cursor has a delay so I need to predict where to aim. I have recorded short gameplay:

https://user-images.githubusercontent.com/64481039/149632671-144aa500-cb1d-4026-bdd1-a475bfdafdba.mp4

Logs that I have checked this time:
[logs.zip](https://github.com/ppy/osu/files/7875521/logs.zip)

I have also checked my graphics drivers and they seem fine. I will try checking with other games.

**Reply by bdach** [reply]
well the fps counter on the video attached looks good for one thing. are you saying that 260fps draw is low for you and you're expecting to see more? it would be over 2x of the refresh rate of most displays, even 144-165hz ones. so there's no real point to running any higher fps than that. you cannot physically see any frames over what the display refresh rate is.

i can't see any good reason why there would be "cursor delay" either, the input thread is running well. what is your input device? do you have anything custom installed (custom drivers, whatever else)? maybe the source of delay is something else like audio?

**Reply by taep96** [reply]
Maybe I mistook game being not smooth because of the cursor? As I said I tried 4 different drivers (built-in kernel wacom driver, in-game otd, external otd, my own driver). This also affects my mouse, not just tablet. Also this doesn't on welcome screen and outside of osu.

**Reply by integerrr** [reply]
Tried exclusive fullscreen yet?

**Reply by taep96** [reply]
[replays.zip](https://github.com/ppy/osu/files/7882497/replays.zip)

**Reply by smoogipoo** [reply]
Hard to say, could be any number of reasons including:
- Kernel
- Windowing system (Wayland or X?)
- Compositor / compositor settings

Replays don't help. There's not much we can do to help you.

**Reply by taep96** [reply]
Tried using zen kernel and killing compositor process. Using X

**Reply by taep96** [reply]
Pop!_OS came with GNOME but on my current system I use dwm instead. Maybe this somehow affects it?

**Reply by taep96** [reply]
Looks like there is less lag on lts kernel, but I think there is still some. Song select is just as laggy

**Reply by taep96** [reply]
Could this not be just the cursor but whole screen? Since only audio is not lagging I don't really have anything else to tell that it's just the cursor. That would also explain why it lags differently in main menu, song select, and when playing a map.

**Reply by bdach** [reply]
your guess is as good as mine. sorry there's not much more help that i can personally provide without owning the actual hardware and being able to debug it directly rather than throw guesses at a wall

**Reply by taep96** [reply]
I guess reporting some guesses wouldn't do any harm? In case someone who knows finds this could help.

I also discovered that this is being logged on each startup:
![image](https://user-images.githubusercontent.com/64481039/151702220-f425cdcd-6e53-415c-a842-f7aebbe6613b.png)
![image](https://user-images.githubusercontent.com/64481039/151702234-02364c76-59be-4c43-b5e9-3d19e18fec64.png)

That's why I came to conclusion whole screen could be lagging.

**Reply by bdach** [reply]
those dmesg entries look rather unrelated.

also i'd say it seems highly unlikely that a screen is changing its latency parameters depending on what is displayed on the screen? depends on what the screen is i guess. it may be more noticeable in gameplay than elsewhere too i guess.

**Reply by taep96** [reply]
After fixing iommu there is no render delay, but cursor is still laggy in map select.

https://user-images.githubusercontent.com/64481039/152013958-31650dfd-23c6-4f57-83dc-5384177275db.mp4

**Reply by taep96** [reply]
Updated the title

**Reply by taep96** [reply]
Settings this options in `/etc/X11/xorg.conf.d/20-amdgpu.conf` solved the issue
```
Section "Device"
     Identifier "AMD"
     Driver "amdgpu"
     Option "EnablePageFlip" "off"
     Option "TearFree" "false"
EndSection
```
